In [2]:
import pandas as pd

In [67]:
# load datasets
datasets = {}
years = range(2018, 2021)
for year in years:
    print(f"Schools stats {year} Loading...")
    file_name = f'ZZSO-{year}.xlsx'
    dataset = pd.read_excel(f"./fin_data/{file_name}", header=1, dtype='str')
    datasets.update({year:dataset})
    print("success")

Schools stats 2018 Loading...
success
Schools stats 2019 Loading...
success
Schools stats 2020 Loading...
success


In [69]:
for year in years:
    print(year,'\n', list(datasets[year].columns))

2018 
 ['№ Області', 'Область', 'Тип територіальної одиниці', '№№', 'Назва тер. Одиниці', 'Найменування закладу', 'Ступінь закладу ( I, I-II, I-III)', 'Загальна площа приміщень школи ( без господарських будівель) (кв.м.)', 'Кількість штатних працівників педагогічного персоналу  разом', 'Кількість штатних працівників непедагогічного персоналу  разом', 'Кількість штатних працівників педагогічного персоналу  пенсійного віку', 'Питома вага непедагогічного персоналу у загальній кількості працівників, %**', 'Питома вага педагогічних працівників пенсійного віку у загальній кількості педагогічних працівників, %**', 'Кількість учнів разом', 'Кількість класів разом', 'Фактична наповнюваність класів, учнів на клас**', 'Показник співвідношення кількості учнів на одного вчителя, учнів на вчителя**', 'Видатки на функціонування закладу (загальний фонд), тис.грн.\n2018рік (касові видатки)', 'Видатки на функціонування закладу (загальний фонд), тис.грн.\n2019 рік (план)', 'Видатки на 1 учня, 2018, тис. 

In [70]:
rename_dict = {
    "Область": "regname",
    "Тип територіальної одиниці": "tertype",
    "Тип школи: міська / сільська": 'eotype',
    "Назва тер. Одиниці": "areaname",
    "Найменування закладу": "eoname",
    "Ступінь закладу ( I, I-II, I-III)": "eolevel",
    "Загальна площа приміщень школи ( без господарських будівель) (кв.м.)": "eosize",
    "Кількість штатних працівників педагогічного персоналу  разом": "teachstuff",
    "Кількість штатних працівників непедагогічного персоналу  разом": "nonteachstuff",
    "Кількість штатних працівників педагогічного персоналу  пенсійного віку": "teachstuffretage",
    "Кількість учнів разом": "pupils",
    "Кількість класів разом": "classes",
    "Разом кількість учнів, та дітей, які здобувають дошкільну освіту": "pupilswithpreschool",
    "Видатки на функціонування закладу (загальний фонд), тис.грн.\n2018рік (касові видатки)": "opex",
    "Видатки на функціонування закладу (загальний фонд), тис.грн.\n2019 рік (касові видатки)": "opex",
    "Видатки на функціонування закладу (загальний фонд), тис.грн.\n2020 рік (касові видатки)": "opex",
    "Видатки на функціонування закладу (загальний фонд), тис.грн.\n2019 рік (план)": "opexplan",
    "Видатки на функціонування закладу (загальний фонд), тис.грн.\n2020 рік (план)": "opexplan",
    "Видатки на функціонування закладу (загальний фонд), тис.грн.\n2021 рік (план)": "opexplan",
    "Опорна": "hub",
    "Код бюджету": "budgetcode",
    "Форма власності": "ownership",
    "Код ДІСО": "DISO",
    "ЄДРПОУ": "EDRPOU"
    }

In [71]:
for year in years:
    datasets[year].rename(columns=rename_dict, inplace=True)
    to_drop = [col for col in datasets[year].columns if col not in rename_dict.values()]
    datasets[year].drop(columns = to_drop, inplace = True)
    datasets[year]['year'] = year

school_finance = pd.concat([datasets[2018], datasets[2019], datasets[2020]], ignore_index=True, sort=False)
school_finance.columns


Index(['regname', 'tertype', 'areaname', 'eoname', 'eolevel', 'eosize',
       'teachstuff', 'nonteachstuff', 'teachstuffretage', 'pupils', 'classes',
       'opex', 'opexplan', 'year', 'eotype', 'pupilswithpreschool', 'hub',
       'budgetcode', 'ownership', 'DISO', 'EDRPOU'],
      dtype='object')

In [72]:
set(school_finance.eolevel.unique())

{' I-III ',
 '0',
 'I',
 'I-II',
 'I-III',
 'I-IIІ',
 'II',
 'II-III',
 'III',
 'ІІ-ІІІ'}

In [74]:
values_to_drop = ['I', 'I-II', 'II']

school_finance = school_finance[~school_finance.eolevel.isin(values_to_drop)]
school_finance.reset_index(drop=True, inplace=True)
school_finance

regname tertype                     areaname  \
0      Дніпропетровська   Район          П'ятихатський район   
1              Донецька   Місто                 м. Маріуполь   
2              Донецька   Район  Великоновосілківський район   
3              Донецька   Район           Волноваський район   
4            Запорізька   Район               Токмацький р-н   
...                 ...     ...                          ...   
28717        місто Київ     NaN                   місто Київ   
28718        місто Київ     NaN                   місто Київ   
28719        місто Київ     NaN                   місто Київ   
28720        місто Київ     NaN                   місто Київ   
28721        місто Київ     NaN                   місто Київ   

                                                  eoname eolevel eosize  \
0      Троїцький навчально- виховний комплекс "загаль...   I-III   2766   
1      Комунальний заклад "Маріупольська загальноосві...   I-III      0   
2      Зеленопільська загальноосвітня школа І-ІІІ сту...   I-III    830   
3      Зачатівська загальноосвітня школа№2 Волноваськ...   I-III      0   
4                                         Ударнецька ЗОШ   I-III      0   
...                                                  ...     ...    ...   
28717                                     гімназія № 237   I-III    NaN   
28718  Середня загальноосвітня школа І-ІІІ ступенів №...   I-III    NaN   
28719  Спеціалізована школа І-ІІІ ступенів № 73 з пог...   I-III    NaN   
28720  Спеціалізована школа № 131 Святошинського райо...   I-III    NaN   
28721  Школа І ступеня № 333 Дарницького району м. Києва   I-III    NaN   

      teachstuff nonteachstuff teachstuffretage pupils  ...  \
0              0             0                0      0  ...   
1              0             4                0      0  ...   
2              0             1                0      0  ...   
3              0           3.5                0      0  ...   
4              0             5                0      0  ...   
...          ...           ...              ...    ...  ...   
28717         52            23                2    905  ...   
28718         58            26                2    827  ...   
28719         32            22                1    483  ...   
28720         50            30                1    861  ...   
28721         29            23                0    585  ...   

                    opex   opexplan  year  eotype pupilswithpreschool  hub  \
0               1175.116          0  2018     NaN                 NaN  NaN   
1                1877.06     469.82  2018     NaN                 NaN  NaN   
2                 50.076   56.08512  2018     NaN                 NaN  NaN   
3                219.893    224.834  2018     NaN                 NaN  NaN   
4                264.315    314.839  2018     NaN                 NaN  NaN   
...                  ...        ...   ...     ...                 ...  ...   
28717            19606.3  25731.415  2020   місто                 NaN  NaN   
28718            22065.4    26469.2  2020   місто                 NaN  NaN   
28719  15793.70295055802  10294.196  2020   місто                 NaN  NaN   
28720            20353.2    13412.1  2020   місто                 NaN  NaN   
28721           11503.71  15556.951  2020   місто                 NaN  NaN   

        budgetcode   ownership   DISO    EDRPOU  
0              NaN         NaN    NaN       NaN  
1              NaN         NaN    NaN       NaN  
2              NaN         NaN    NaN       NaN  
3              NaN         NaN    NaN       NaN  
4              NaN         NaN    NaN       NaN  
...            ...         ...    ...       ...  
28717  26000000000  комунальна  22883  23392178  
28718  26000000000  комунальна  23029  23392149  
28719  26000000000  комунальна  23280  26187881  
28720  26000000000  комунальна  23154  22878447  
28721  26000000000  комунальна  23680  41659088  

[28722 rows x 21 columns]

In [83]:
school_finance[school_finance.EDRPOU.str.strip().str.len() != 8].EDRPOU.unique()

array([nan, '62059', '7995', '0', '11'], dtype=object)

In [82]:
school_finance.EDRPOU = school_finance.EDRPOU.apply(lambda x: x.zfill(8) if isinstance(x, str) and len(x) >= 6 else x)

In [92]:
school_finance[(school_finance.EDRPOU.str.strip().str.len() != 8) & (school_finance.EDRPOU.notna())][['eoname','EDRPOU']]

eoname EDRPOU
19602  Турбівська загальноосвітня школа І-ІІІ ст. №2 ...  62059
19608  Середня загальноосвітня школа І - ІІІ ступенів...   7995
19612  Середня загальноосвітня школа І-ІІІ ступенів с...      0
19833  Комунальний заклад "Загальноосвітня школа І-ІІ...     11
23424  Старобільська загальноосвітня школа ІІ-ІІІ сту...      0

# STOP

In [6]:
datasets[2019].loc[datasets[2019]['EDRPOU'].apply(len) == 7, 'EDRPOU'] = '0' + datasets[2019].loc[datasets[2019]['EDRPOU'].apply(len) == 7, 'EDRPOU']
datasets[2019].loc[datasets[2019]['eoname'] == 'Марянівський навчально-виховний комплекс "ДНЗ-ЗОШ" Іступеня', 'EDRPOU'] = '26224046'
datasets[2019].loc[datasets[2019]['eoname'] == 'Лобойківський заклад загальної середньої освіти Петриківської районної ради Дніпропетровської області', 'EDRPOU'] = '26460466'
datasets[2019].loc[datasets[2019]['eoname'] == 'Середня загальноосвітня школа І-ІІ ступенів села Вартиківці Кельменецького району Чернівецької області', 'EDRPOU'] = '31967606'
datasets[2019].loc[datasets[2019]['eoname'] == 'Рипенська загальноосвітня школа І ступеня Путильської районної ради Путильський район Чернівецька область', 'EDRPOU'] = '25811521'
datasets[2019].loc[datasets[2019]['eoname'] == 'Рижівська загальноосвітня школа І ступеня села Рижа Путильського району Чернівецької області', 'EDRPOU'] = '25811484'
datasets[2019].loc[datasets[2019]['eoname'] == 'Великобушинська загальноосвітня школа І-ІІ ступенів Немирівської районної ради Вінницької області', 'EDRPOU'] = '25513410'
datasets[2019].loc[datasets[2019]['eoname'] == 'Мирненська загальноосвітня школа І ступеня Надвірнянської районної ради Івано-Франківської області', 'EDRPOU'] = '20567542'
datasets[2019].loc[datasets[2019]['eoname'] == 'Завадівський навчально-виховний комплекс І-ІІ ступенів Турківської районної ради Львівської обасті', 'EDRPOU'] = '22417999'
datasets[2019].loc[datasets[2019]['eoname'] == 'Херсонський навчально-виховний комплекс "Загальноосвітня школа ІІ ступеня-ліцей журналістики, економіки та правознавства" Херсонської міської ради', 'EDRPOU'] = '24114961'

In [9]:
datasets[2019][datasets[2019].duplicated(subset=['EDRPOU'], keep=False)]

eoname      regname  \
53     Марянівський навчально-виховний комплекс "ДНЗ-...    Вінницька   
489    Загальноосвітня школа І-ІІІ ступенів с.Мар'янівка    Вінницька   
6336   Бориславська ЗОШ І-ІІІ ступенів №8 Бориславськ...    Львівська   
6357   Навчально-виховний комплекс імені Володимира Т...    Львівська   
6359   Новороздільська загальноосвітня школа І-ІІІ ст...    Львівська   
...                                                  ...          ...   
9640   Вараська загальноосвiтня школа I-III ступенiв ...   Рівненська   
13106  Путильська загальноосвітня школа І-ІІІ ступені...  Чернівецька   
13110  Сергіївська загальноосвітня школа І-ІІІ ступен...  Чернівецька   
13113  Рипенська загальноосвітня школа І ступеня Пути...  Чернівецька   
13114  Рижівська загальноосвітня школа І ступеня села...  Чернівецька   

                  areaname eolevel eosize teachstuff nonteachstuff  \
53          Барський район       I    NaN          7            10   
489     Хмільницький район   I-III    NaN         13             9   
6336        місто Борислав   I-III    NaN         38            12   
6357    місто Новий Розділ   I-III    NaN         57            23   
6359    місто Новий Розділ   I-III    NaN         47            17   
...                    ...     ...    ...        ...           ...   
9640   Вараська міська ОТГ   I-III    NaN         85            35   
13106    Путильський район   I-III    NaN         64            29   
13110    Путильський район   I-III    NaN         34            15   
13113    Путильський район       I    NaN          5             1   
13114    Путильський район       I    NaN          5             1   

      teachstuffretage pupils pupilswithpreschool classes     opex opexplan  \
53                   0     15                  35       1   1591.8   1391.1   
489                  1     60                  60       7   2648.4   2848.7   
6336                 6    478                 478      19   7823.4   7636.5   
6357                13    474                 474      20     9880    10671   
6359                10    521                 521      21     8148     8872   
...                ...    ...                 ...     ...      ...      ...   
9640                24    819                 819      32  17044.4    19336   
13106                7    662                 662      27  13568.4  14271.9   
13110                9    287                 287      14   7263.1   7682.8   
13113                0     18                  18       2    717.1    750.8   
13114                0     21                  21       2    723.9    767.3   

      hub   budgetcode   ownership   DISO    EDRPOU  year  
53     ні   2301200000  комунальна  22767  26224046  2019  
489    ні   2323200000  комунальна   8463  26224046  2019  
6336   ні  13202100000  комунальна  12302  02144795  2019  
6357   ні  13205100000  комунальна  12957  26454595  2019  
6359   ні  13205100000  комунальна  12959  26454595  2019  
...    ..          ...         ...    ...       ...   ...  
9640   ні  17532000000  комунальна  16449  04590731  2019  
13106  ні  24308200000  комунальна   5585  25811484  2019  
13110  ні  24308200000  комунальна   5614  25811521  2019  
13113  ні  24308200000  комунальна   6116  25811521  2019  
13114  ні  24308200000  комунальна   6338  25811484  2019  

[72 rows x 19 columns]

Якщо подивитися деякі з цих ЄДРПОУ, бачимо таке:

26224046 це Марянівський навчально-виховний комплекс, а друга школа з цим єдрпоу взагалі немає єдрпоу в реєстрах

02144795 це Головне управління освіти і науки львівської обласної державної адміністрації

25811521 це Сергіївська загальноосвітня школа, а Рипенська загальноосвітня школа І ступеня це філіал, який має свій єдрпоу і т.д.

In [11]:
datasets[2020].loc[datasets[2020]['EDRPOU'].apply(len) == 7, 'EDRPOU'] = '0' + datasets[2020].loc[datasets[2020]['EDRPOU'].apply(len) == 7, 'EDRPOU']
datasets[2020].loc[datasets[2020]['eoname'] == 'Турбівська загальноосвітня школа І-ІІІ ст. №2 _x000D_\nЛиповецької районної ради _x000D_\nВінницької області_x000D_\n_x000D_\n', 'EDRPOU'] = '21723820'
datasets[2020].loc[datasets[2020]['eoname'] == 'Середня загальноосвітня школа І - ІІІ ступенів села Уладівка Літинський район Вінницька область', 'EDRPOU'] = '26229088'
datasets[2020].loc[datasets[2020]['eoname'] == 'Середня загальноосвітня школа І-ІІІ ступенів с. Івча', 'EDRPOU'] = '26223660'
datasets[2020].loc[datasets[2020]['eoname'] == 'Комунальний заклад "Загальноосвітня школа І-ІІІ ступенів № 19 Вінницької міської ради"', 'EDRPOU'] = '26235447'
datasets[2020].loc[datasets[2020]['eoname'] == 'Тисаашванська загальноосвітня школа І-ІІ ступенів Ужгородської районної ради Закарпатської області', 'EDRPOU'] = '22103539'
datasets[2020].loc[datasets[2020]['eoname'] == 'Пшеничниківський навчально-виховний комплекс "Загальноосвітня школа І-ІІ ступенів - дошкільний навчальний заклад" Тисменицької районної ради Івано-Франківської області', 'EDRPOU'] = '20563076'
datasets[2020].loc[datasets[2020]['eoname'] == 'Старобільська загальноосвітня школа ІІ-ІІІ ступенів №4 Старобільської районної ради Луганської області', 'EDRPOU'] = '33199525'
datasets[2020].loc[datasets[2020]['eoname'] == 'Заклад загальноосвітньої середньої освіти І ступення села Смеречка Старосамбірського району Львівської області', 'EDRPOU'] = '00000000'
datasets[2020].loc[datasets[2020]['eoname'] == 'Загальноосвітня середня школа І ступеня села Дністрик Старосамбірського району Львівської області', 'EDRPOU'] = '00000000'
datasets[2020].loc[datasets[2020]['eoname'] == 'Заклад загальної середньої освіти І ступеня села Головецько-Золотнівець Старосамбірського району Львівської області', 'EDRPOU'] = '00000000'
datasets[2020].loc[datasets[2020]['eoname'] == "Воле - Блажівська середня загальноосвітня школа І ступеня Чукв'янської сільської ради Самбірського району Львівської області", 'EDRPOU'] = '22378609'
datasets[2020].loc[datasets[2020]['eoname'] == 'Загальноосвітня школа І ступеня села Виноградне', 'EDRPOU'] = '24625450'
datasets[2020].loc[datasets[2020]['eoname'] == ' Херсонський навчально-виховний комплекс "Загальноосвітня школа ІІ ст-ліцей журналістики, економіки та правознавства" Херсонської міської ради', 'EDRPOU'] = '24114961'

In [12]:
datasets[2020][datasets[2020].duplicated(subset=['EDRPOU'], keep=False)]

eoname      regname  \
17     Каришківська загальноосвітня школа І-ІІІ ступенів    Вінницька   
27     Войнашівська загальноосвітня школа І-ІІІ ступе...    Вінницька   
37             Лісівська загальноосвітня школа І ступеня    Вінницька   
47     Бершадська загальноосвітня школа І-ІІІ ступені...    Вінницька   
60           Флоринська загальноосвітня школа І -ІІІ ст.    Вінницька   
...                                                  ...          ...   
12610  Вербовецький заклад загальної середньої освіти...  Чернівецька   
13227     Заклад загальної середньої освіти №223 м.Києва   місто Київ   
13290  спеціалізована школа № 317 Святошинського райо...   місто Київ   
13356  м.Київ, середня загальноосвітня школа № 162, С...   місто Київ   
13381                                Гімназія "Академія"   місто Київ   

                        areaname eolevel eosize teachstuff nonteachstuff  \
17     Копайгородська селищна ТГ   I-III    NaN         20            16   
27             Барcька міська ТГ   I-III    NaN         45            25   
37     Копайгородська селищна ТГ       I    NaN          5             4   
47          Бершадська міська ТГ   I-III    NaN         54            30   
60          Бершадська міська ТГ   I-III    NaN         30            19   
...                          ...     ...    ...        ...           ...   
12610    Заставнівська міська ТГ    I-II    NaN         11             5   
13227                 місто Київ   I-III    NaN         48            25   
13290                 місто Київ   I-III    NaN         64            30   
13356                 місто Київ   I-III    NaN         60            35   
13381                 місто Київ   I-III    NaN         68            40   

      teachstuffretage pupils classes       opex   opexplan     hub  \
17                   4     87      11       4441     5847.5     NaN   
27                   5    410      27    11504.7      13843  опорна   
37                   0     23       3       1034       1533     NaN   
47                   9    608      28  13942.865  17284.594     NaN   
60                   2    373      20   8332.107   9907.135     NaN   
...                ...    ...     ...        ...        ...     ...   
12610                0     57       7       2155       2713     NaN   
13227               10    946      33    23968.2    16224.7     NaN   
13290               11   1175      42    27892.2    20529.9     NaN   
13356                8   1072      37    25588.9    19150.6     NaN   
13381                8   1217      39    30102.7    21184.7     NaN   

        budgetcode   ownership   DISO    EDRPOU  year  
17     02554000000  комунальна   6927  02141288  2020  
27     02505000000  комунальна   6875  02141288  2020  
37     02554000000  комунальна   6918  02141288  2020  
47     02548000000  комунальна   6152  02141294  2020  
60     02548000000  комунальна   6144  02141294  2020  
...            ...         ...    ...       ...   ...  
12610  24528000000  комунальна   5552  22850549  2020  
13227  26000000000  комунальна  23168  37498536  2020  
13290  26000000000  комунальна  23178  37498536  2020  
13356  26000000000  комунальна  23157  37498536  2020  
13381  26000000000  комунальна  23179  37498536  2020  

[109 rows x 18 columns]

In [19]:
concatenated_df = pd.concat(list(datasets.values()), ignore_index=True)
concatenated_df

eoname eolevel  eosize  \
0      КЗ «Ординецька ЗОШ І ступеня Погреб районної р...       I     888   
1      Олександрівська ЗОШ І ступеня Магдалинівської ...       I  1467.4   
2      Крамарська загальноосвітня школа І-ІІ ступенів...    I-II  524.21   
3      Минівська ЗОШ І ступеня Заплавської сільської ...       I     881   
4      Новотроїцька загальноосвітня школа I- II ступе...    I-II   205.7   
...                                                  ...     ...     ...   
42404  Спеціалізований навчально-виховний комплекс "К...       I     NaN   
42405  НВК "Спеціалізована школа І ступеня з поглибле...       I     NaN   
42406  Початкова школа № 334 Дарницького району м. Києва       I     NaN   
42407  Школа І ступеня № 333 Дарницького району м. Києва   I-III     NaN   
42408   Початкова школа №332 Дарницького району м. Києва       I     NaN   

      teachstuff nonteachstuff teachstuffretage pupils classes        opex  \
0              0             0                0      0       0           0   
1              0             4                0      0       0     237.494   
2              0             2                0      0       0     821.143   
3              0             3                0      0       0     343.575   
4              0             0                0      0       0     927.012   
...          ...           ...              ...    ...     ...         ...   
42404         11            21                0    166       6  5415.94732   
42405         14            21                0    210       7    9401.191   
42406         24            34                0    449      14     9552.78   
42407         29            23                0    585      16    11503.71   
42408         44            44                0   1017      29    17969.26   

        opexplan  year pupilswithpreschool  hub   budgetcode   ownership  \
0              0  2018                 NaN  NaN          NaN         NaN   
1              0  2018                 NaN  NaN          NaN         NaN   
2              0  2018                 NaN  NaN          NaN         NaN   
3              0  2018                 NaN  NaN          NaN         NaN   
4              0  2018                 NaN  NaN          NaN         NaN   
...          ...   ...                 ...  ...          ...         ...   
42404     6742.1  2020                 NaN  NaN  26000000000  комунальна   
42405   11418.88  2020                 NaN  NaN  26000000000  комунальна   
42406  15143.279  2020                 NaN  NaN  26000000000  комунальна   
42407  15556.951  2020                 NaN  NaN  26000000000  комунальна   
42408  26466.605  2020                 NaN  NaN  26000000000  комунальна   

        DISO Код ЄДРПОУ  
0        NaN        NaN  
1        NaN        NaN  
2        NaN        NaN  
3        NaN        NaN  
4        NaN        NaN  
...      ...        ...  
42404  22986   22874627  
42405  23112   21580289  
42406  23731   42384408  
42407  23680   41659088  
42408  23730   42384345  

[42409 rows x 17 columns]

In [35]:
concatenated_df[concatenated_df['Код ЄДРПОУ'].notna()][concatenated_df[concatenated_df['Код ЄДРПОУ'].notna()]['Код ЄДРПОУ'].apply(len) != 8]

eoname eolevel eosize  \
14828  Каришківська загальноосвітня школа І-ІІІ ступенів   I-III    NaN   
14829  Марянівський навчально-виховний комплекс "ДНЗ-...       I    NaN   
15000  Навчально- виховний комплекс "Загальноосвітній...   I-III    NaN   
15012  Зозівська загальноосвітня школа І-ІІІ ст. Липо...   I-III    NaN   
15075       Григорівська загальноосвітня школа 1 ступеня       I    NaN   
...                                                  ...     ...    ...   
42303  Спеціалізована школа І-ІІІ ступенів №152 з пог...   I-III    NaN   
42311  Школа І-ІІІ ступенів № 294 Десянського району ...   I-III    NaN   
42345  Школа І-ІІІ ступенів № 282 Деснянського району...   I-III    NaN   
42381  Спеціалізована школа І-ІІІ ступенів №202 з пог...   I-III    NaN   
42390  Спеціалізована школа І-ІІІ ступенів № 23 з пог...   I-III    NaN   

      teachstuff nonteachstuff teachstuffretage pupils classes  \
14828         22            17                5     87      11   
14829          7            10                0     15       1   
15000         17            13                2     61       8   
15012         22            18                5    174      10   
15075          3             2                0     17       3   
...          ...           ...              ...    ...     ...   
42303         44            30                6    588      24   
42311         62            33                8   1132      41   
42345         55            29                6    973      33   
42381         48            29                3    732      29   
42390         49            26                2    723      24   

                     opex opexplan  year pupilswithpreschool  hub  \
14828              3980.4   4541.4  2019                  87   ні   
14829              1591.8   1391.1  2019                  35   ні   
15000              3744.6   4221.4  2019                  68   ні   
15012              4798.8   5075.6  2019                 174   ні   
15075             530.214  712.694  2019                  17   ні   
...                   ...      ...   ...                 ...  ...   
42303         18971.00334  22593.8  2020                 NaN  NaN   
42311  27706.449149999997  31025.2  2020                 NaN  NaN   
42345         23451.75618  29592.3  2020                 NaN  NaN   
42381  18535.171670000003    22414  2020                 NaN  NaN   
42390  26648.166719999997  31694.8  2020                 NaN  NaN   

        budgetcode   ownership   DISO Код ЄДРПОУ  
14828   2301200000  комунальна   6927    2141288  
14829   2301200000  комунальна  22767          0  
15000   2309200000  комунальна  11377    2141360  
15012   2310200000  комунальна  11395    2141377  
15075   2312200000  комунальна   7183    4326572  
...            ...         ...    ...        ...  
42303  26000000000  комунальна  22933    3875118  
42311  26000000000  комунальна  22960    1489305  
42345  26000000000  комунальна  22956    1488067  
42381  26000000000  комунальна  22937    3875354  
42390  26000000000  комунальна  22929    3874432  

[473 rows x 17 columns]

In [ ]:
# concatenated_df[['pupilswithpreschool', 'budgetcode', 'DISO', 'Код ЄДРПОУ']] = concatenated_df[['pupilswithpreschool', 'budgetcode', 'DISO', 'Код ЄДРПОУ']].fillna(-10000)
# concatenated_df = concatenated_df.astype({'teachstuff': 'int64', 'nonteachstuff': 'int64', 'teachstuffretage': 'int64', 'pupilswithpreschool': 'int64', 'budgetcode': 'int64', 'DISO': 'int64', 'Код ЄДРПОУ': 'int64'})
# concatenated_df = concatenated_df.astype({'pupilswithpreschool': 'str', 'budgetcode': 'str', 'DISO': 'str', 'Код ЄДРПОУ': 'str'})
# concatenated_df[['pupilswithpreschool', 'budgetcode', 'DISO', 'Код ЄДРПОУ']] = concatenated_df[['pupilswithpreschool', 'budgetcode', 'DISO', 'Код ЄДРПОУ']].replace('-10000', np.NaN)
# concatenated_df

In [36]:
concatenated_df.to_csv('Scools_stats.csv', index=False, encoding='utf-8')